src/main/java/org/thingsboard/gateway/extensions/sigfox/DefaultSigfoxService.java

```
@@ -0,0 +1,82 @@
+
+    @Autowired
+    private GatewayService gateway;
+
+    private List<SigfoxDeviceTypeConfiguration> sigfoxDeviceTypeConfigurations;
+
+    @PostConstruct
+    public void init() throws IOException {
+        SigfoxConfiguration configuration;
+        try {
+            configuration = ConfigurationTools.readConfiguration(configurationFile, SigfoxConfiguration.class);
+        } catch (IOException e) {
+            log.error("Sigfox service configurataion failed!", e);
+            throw e;
+        }
+        this.sigfoxDeviceTypeConfigurations = configuration.getDeviceTypes();
+    }
+
+    @Override
+    public void processRequest(String deviceTypeId, String token, String body) throws Exception {
+        Optional<SigfoxDeviceTypeConfiguration> configuration = findDeviceTypeConfigurationByDeviceTypeId(deviceTypeId);
+        if (configuration.isPresent()) {
+            final SigfoxDeviceTypeConfiguration deviceTypeConfiguration = configuration.get();
+            if (deviceTypeConfiguration.getToken().equals(token)) {
+                processJson(body, deviceTypeConfiguration);
+            } else {
+                log.error("Request token [{}] for device type id [{}] doesn't match configuration token!", token, deviceTypeId);
+                throw new SecurityException("Request token ["+ token + "] for device type id [" + deviceTypeId + "] doesn't match configuration token!");
+
+            }
+        } else {
+            log.error("No configuration found for device type id [{}]. Please add configuration for this device type id!", deviceTypeId);
+            throw new IllegalArgumentException("No configuration found for device type id [" + deviceTypeId + "]. Please add configuration for this device type id!");
+        }
+    }
+
+    private void processJson(String body, SigfoxDeviceTypeConfiguration deviceTypeConfiguration) {
+        DeviceData dd = deviceTypeConfiguration.getMapping().parseBody(body);
+        if (dd != null) {
```

if dd == null, maybe let's raise an exception?